modules/mono/glue/cs_files/Mathf.cs

```diff
@@ -139,28 +167,57 @@ public static real_t Floor(real_t s)
             return (real_t)Math.Floor(s);
         }
 
-        public static real_t Fposmod(real_t x, real_t y)
+        public static int FloorToInt(real_t s)
         {
-            if (x >= 0f)
-            {
-                return x % y;
-            }
-            else
+            return (int)Math.Floor(s);
+        }
+
+        public static real_t Fposmod(real_t a, real_t b)
+        {
+            real_t c = a % b;
+            if (c < 0) 
             {
-                return y - (-x % y);
+                c += b;
             }
+            return c;
         }
 
         public static real_t Lerp(real_t from, real_t to, real_t weight)
         {
             return from + (to - from) * Clamp(weight, 0f, 1f);
         }
 
+        /// <summary>
+        /// This returns the Logarithm with base E (Natural Log)
+        /// </summary>
         public static real_t Log(real_t s)
         {
             return (real_t)Math.Log(s);
         }
 
+        /// <summary>
+        /// This returns the Logarithm with base `b` 
+        /// </summary>
+        public static real_t LogBase(real_t a, real_t b)
+        {
+            return (real_t)Math.Log(a, b);
+        }
+
+        public static real_t Log10(real_t s)
+        {
+            return (real_t)Math.Log10(s);
+        }
+
+        public static real_t Log2(real_t s)
+        {
+            return (real_t)Math.Log(s, 2);
+        }
+
+        public static real_t LogE(real_t s)
+        {
+            return (real_t)Math.Log(s);
```

Comment:
Why you create two functions with same functionality(LogE and Log) ?

Reply:
`Log` for backwards compatibility and the fact that it's the same name as in System.Math, but I prefer the term `LogE` as it's more clear and follows the pattern of `Log10` and `Log2`.

Reply:
And its not be confusable for other users ?

Reply:
LogE is a less ambiguous term. People looking for natural log will be able to easily find it with LogE, people looking for just any log or a wrapper for Math.Log will look for Mathf.Log.

Lots of these changes mostly improve the readability of a game's code.